src/Controller/Controller.php

```diff
@@ -179,23 +179,19 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * Sets a number of properties based on conventions if they are empty. To override the
      * conventions CakePHP uses you can define properties in your class declaration.
      *
-     * @param \Cake\Http\ServerRequest|null $request Request object for this controller. Can be null for testing,
+     * @param \Cake\Http\ServerRequest $request Request object for this controller.
      *   but expect that features that use the request parameters will not work.
-     * @param \Cake\Http\Response|null $response Response object for this controller.
      * @param string|null $name Override the name useful in testing when using mocks.
      * @param \Cake\Event\EventManagerInterface|null $eventManager The event manager. Defaults to a new instance.
-     * @param \Cake\Controller\ComponentRegistry|null $components The component registry. Defaults to a new instance.
      */
     public function __construct(
-        ?ServerRequest $request = null,
-        ?Response $response = null,
```

Comment:
We'll need to document this as I bet somewhere userland code is relying on this. I do like the the improved simplicity of this though.

Reply:
I'll open a PR for the migration guide.

Reply:
This will affect users who inject controllers from a service container. They have to add a constructor to their controllers and call parent::__construct() with these parameters.